lambdas/process-journey-event/src/main/java/uk/gov/di/ipv/core/processjourneyevent/statemachine/StateMachine.java

```diff
@@ -27,6 +38,28 @@ public State transition(String startState, String event, JourneyContext journeyC
                     String.format("Unknown state provided to state machine: %s", startState));
         }
 
+        if (currentPage.isPresent() && !currentPage.get().equals("oauth")) {
+            if (state instanceof BasicState) {
+
+                if (((BasicState) state).getResponse() instanceof PageStepResponse) {
+
+                    String pageId =
+                            ((PageStepResponse) ((BasicState) state).getResponse()).getPageId();
+
+                    if (!currentPage.get().equals(pageId)) {
+                        return state;
```

Comment:
Yeah I'm sure it will work for page requests, we are basically saying if the current backend page is not the same as the current front-end page, ignore what the front-end is asking us to do and just take the user to where the backend thinks we are.

Then the front-end will set the current page correctly and we will be back in sync.

and fairly confident on CRI pages (although I've only been able to test it with the stubs, not the actual CRI flow so it's possibly worth disabled the transition above staging to test it thoroughly before we hit production)

With lambda function calls we will need to error the journey, there's not really much more we can do here I don't think